data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerMessageHandler.java

```diff
@@ -109,6 +110,13 @@ private void reconcileLast(final DataPlaneContract.Contract newContract) {
           // If that's the case, reconcile it.
           if (last.get().getGeneration() != contract.getGeneration()) {
             reconcileLast(null);
+          } else if (r.failed()) {
+            // Retry failed reconciliations after a few seconds.
+            // We add a delay to avoid looping too much on errors.
+            vertx.setTimer(RECONCILE_FAILED_RETRY_DELAY, v -> {
+              logger.info("Retrying previously failed reconciliation");
+              reconcileLast(null);
```

Comment:
Would this cause the reconciler to try forever every 5 seconds?

Reply:
No, see docs for setTimer:
```
  /**
   * Set a one-shot timer to fire after {@code delay} milliseconds, at which point {@code handler} will be called with
   * the id of the timer.
   *
   * @param delay  the delay in milliseconds, after which the timer will fire
   * @param handler  the handler that will be called with the timer ID when the timer fires
   * @return the unique ID of the timer
   */
```

Reply:
it's called only once

Reply:
kk

/lgtm
/approve